Makefile

```diff
@@ -190,27 +190,26 @@ unit-sup: build-launcher-for-supervisor-tests
 .PHONY: build-launcher-for-supervisor-tests
 
 # Lints we need to work through and decide as a team whether to allow or fix
-UNEXAMINED_LINTS = clippy::cyclomatic_complexity \
-                   clippy::large_enum_variant \
-                   clippy::len_without_is_empty \
-                   clippy::module_inception \
-                   clippy::needless_pass_by_value \
-                   clippy::needless_return \
-                   clippy::new_ret_no_self \
-                   clippy::new_without_default \
-                   clippy::new_without_default_derive \
-                   clippy::question_mark \
-                   clippy::redundant_field_names \
-                   clippy::too_many_arguments \
-                   clippy::trivially_copy_pass_by_ref \
-                   clippy::wrong_self_convention \
-                   renamed_and_removed_lints
+UNEXAMINED_LINTS =
 
 # Lints we disagree with and choose to keep in our code with no warning
-ALLOWED_LINTS =
+ALLOWED_LINTS = clippy::module_inception \
+                clippy::needless_return \
+                clippy::new_ret_no_self \
+                clippy::new_without_default \
```

Comment:
https://rust-lang-nursery.github.io/rust-clippy/master/index.html#new_without_default

I think implementing `Default` should be an affirmative choice, otherwise it's adding complexity unnecessarily and could undermine type-safety by allowing a type to be used where it's not expected merely because it has a default.

[new_without_default.txt](https://github.com/habitat-sh/habitat/files/2813939/new_without_default.txt)

Reply:
I don't know... I think a `new() -> Self` is signaling the same thing as `Default`... I think this lint is basically acknowledging that, and ensuring that your types work properly in the standard Rust API.

If we were to adopt this, it probably should go along with adopting `new_ret_no_self` and `new_without_default_derive`.

Reply:
The main difference is that having a `new` isn't implementing a trait and so doesn't opt-in to special handling by the compiler.

Reply:
Seems like `new_ret_no_self` and this should go hand in hand.

Reply:
No strong opinions here.